client/src/push-content.ts

```diff
@@ -284,16 +281,24 @@ export const pushContent = (hostContext: ExtensionHostContext) => async () => {
       // fix ids
       // TODO: better ui in future. Add `increment` value in `progress.report` and use a callback to update real progress
       await requestEnsureIds(hostContext.client, { workspaceUri: uri.toString() })
-      const privateSubmodule = _getPrivateSubmodule()
-      if (privateSubmodule !== undefined) {
-        try {
+      try {
+        const privateSubmodule = _getPrivateSubmodule()
+        if (privateSubmodule !== undefined) {
           await preparePrivateSubmodule(hostContext.client, privateSubmodule)
           // Private submodule should go first because that means the latest
-          // submodule version will be committed
+          // submodule version will be committed to the book repository
           pushTargets.unshift(privateSubmodule)
-        } catch (e) {
-          const err = e as Error
-          await vscode.window.showErrorMessage(err.message)
+        }
+      } catch (e) {
+        const err = e as Error
+        // Add '.' if the message does not end in '.', '?', or '!'.
+        const errMsg = err.message.trimEnd().replace(/[^.?!]$/, (s) => `${s}.`)
+        const response = await vscode.window.showErrorMessage(
+          `${errMsg} Your changes will not be committed/pushed to the private submodule!`,
```

Comment:
Could append it to 'Your changes will not..." rather than prepending to that string? Maybe something like:

```suggestion
          `Your changes will not be committed/pushed to the private submodule!\nUnderlying reason:\n${errMsg}`,
```

Reply:
Hmm, putting the conclusion at the beginning seems strange to me, but I can see your point about how it could cause problems with localization.